examples/layers/raw/canvas.dart

```diff
@@ -2,19 +2,30 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// This example shows how to use the ui.Canvas interface to draw various shapes
+// with graidents and transforms.
+
 import 'dart:ui' as ui;
 import 'dart:math' as math;
 import 'dart:typed_data';
 
 ui.Picture paint(ui.Rect paintBounds) {
+  // First we create a PictureRecorder to record the commands we're going to
+  // feed in the canvas. The PictureRecorder will evetually produce a Picture,
+  // which is an immutable record of those commands.
   ui.PictureRecorder recorder = new ui.PictureRecorder();
+
+  // Next, we create a canvas from the recorder. The canvas is an interface
+  // which can receive drawing commands.  The canvas interface is modeled after
+  // the SkCanvas interface from Skia.
   ui.Canvas canvas = new ui.Canvas(recorder, paintBounds);
   ui.Size size = paintBounds.size;
 
   ui.Paint paint = new ui.Paint();
+  canvas.drawPaint(new ui.Paint()..color = const ui.Color(0xFFFFFFFF));
+
   ui.Point mid = size.center(ui.Point.origin);
   double radius = size.shortestSide / 2.0;
-  canvas.drawPaint(new ui.Paint()..color = const ui.Color(0xFFFFFFFF));
 
   canvas.save();
   canvas.translate(-mid.x/2.0, ui.window.size.height*2.0);
```

Comment:
the ui.LayerDrawLooperBuilder logic could do with some explanatory comments, it looks intimidating